File: gesttravaux-java/src/main/java/com/immosync/gesttravaux/models/Proprietaire.java
package com.immosync.gesttravaux.models;

import jakarta.persistence.*;
import java.util.HashSet;
import java.util.Set;

/**
 * Entity representing a property owner.
 * A proprietaire can own multiple properties (biens).
 *
 * @author GestTravaux Pro
 * @version 1.0
 */
@Entity
@Table(name = "PROPRIETAIRES")
public class Proprietaire {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "idProprietaire")
    private Integer idProprietaire;

    @Column(name = "nomProprietaire", length = 45, nullable = false)
    private String nomProprietaire;

    @Column(name = "prenomProprietaire", length = 45, nullable = false)
    private String prenomProprietaire;

    @Column(name = "emailProprietaire", length = 45, nullable = false)
    private String emailProprietaire;

    @Column(name = "telProprietaire", length = 45)
    private String telProprietaire;

    @OneToMany(mappedBy = "proprietaire", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private Set<Bien> biens = new HashSet<>();

    /**
     * Default constructor required by JPA.
     */
    public Proprietaire() {
    }

    /**
     * Constructor with essential fields.
     *
     * @param nomProprietaire owner last name
     * @param prenomProprietaire owner first name
     * @param emailProprietaire owner email
     */
    public Proprietaire(String nomProprietaire, String prenomProprietaire, String emailProprietaire) {
        this.nomProprietaire = nomProprietaire;
        this.prenomProprietaire = prenomProprietaire;
        this.emailProprietaire = emailProprietaire;
    }

    // Getters and Setters

    /**
     * Gets the proprietaire ID.
     *
     * @return the proprietaire ID
     */
    public Integer getIdProprietaire() {
        return idProprietaire;
    }

    /**
     * Sets the proprietaire ID.
     *
     * @param idProprietaire the proprietaire ID
     */
    public void setIdProprietaire(Integer idProprietaire) {
        this.idProprietaire = idProprietaire;
    }

    /**
     * Gets the proprietaire last name.
     *
     * @return the last name
     */
    public String getNomProprietaire() {
        return nomProprietaire;
    }

    /**
     * Sets the proprietaire last name.
     *
     * @param nomProprietaire the last name
     */
    public void setNomProprietaire(String nomProprietaire) {
        this.nomProprietaire = nomProprietaire;
    }

    /**
     * Gets the proprietaire first name.
     *
     * @return the first name
     */
    public String getPrenomProprietaire() {
        return prenomProprietaire;
    }

    /**
     * Sets the proprietaire first name.
     *
     * @param prenomProprietaire the first name
     */
    public void setPrenomProprietaire(String prenomProprietaire) {
        this.prenomProprietaire = prenomProprietaire;
    }

    /**
     * Gets the proprietaire email.
     *
     * @return the email
     */
    public String getEmailProprietaire() {
        return emailProprietaire;
    }

    /**
     * Sets the proprietaire email.
     *
     * @param emailProprietaire the email
     */
    public void setEmailProprietaire(String emailProprietaire) {
        this.emailProprietaire = emailProprietaire;
    }

    /**
     * Gets the proprietaire phone number.
     *
     * @return the phone number
     */
    public String getTelProprietaire() {
        return telProprietaire;
    }

    /**
     * Sets the proprietaire phone number.
     *
     * @param telProprietaire the phone number
     */
    public void setTelProprietaire(String telProprietaire) {
        this.telProprietaire = telProprietaire;
    }

    /**
     * Gets the properties owned by this proprietaire.
     *
     * @return set of biens
     */
    public Set<Bien> getBiens() {
        return biens;
    }

    /**
     * Sets the properties owned by this proprietaire.
     *
     * @param biens set of biens
     */
    public void setBiens(Set<Bien> biens) {
        this.biens = biens;
    }

    /**
     * Adds a property to this proprietaire.
     *
     * @param bien the property to add
     */
    public void addBien(Bien bien) {
        biens.add(bien);
        bien.setProprietaire(this);
    }

    /**
     * Removes a property from this proprietaire.
     *
     * @param bien the property to remove
     */
    public void removeBien(Bien bien) {
        biens.remove(bien);
        bien.setProprietaire(null);
    }

    /**
     * Gets the full name of the proprietaire.
     *
     * @return full name (first name + last name)
     */
    public String getFullName() {
        return prenomProprietaire + " " + nomProprietaire;
    }

    @Override
    public String toString() {
        return getFullName();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Proprietaire)) return false;
        Proprietaire that = (Proprietaire) o;
        return idProprietaire != null && idProprietaire.equals(that.getIdProprietaire());
    }

    @Override
    public int hashCode() {
        return getClass().hashCode();
    }
}

File: gesttravaux-symfony/src/Entity/Proprietaire.php
<?php

namespace App\Entity;

use App\Repository\ProprietaireRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;

/**
 * Proprietaire entity - Property owners
 *
 * Represents property owners who request renovation/construction work
 *
 * @author GestTravaux Pro
 */
#[ORM\Entity(repositoryClass: ProprietaireRepository::class)]
#[ORM\Table(name: 'proprietaires')]
class Proprietaire
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(name: 'idProprietaire')]
    private ?int $id = null;

    #[ORM\Column(length: 45)]
    private ?string $nomProprietaire = null;

    #[ORM\Column(length: 45)]
    private ?string $prenomProprietaire = null;

    #[ORM\Column(length: 45)]
    private ?string $emailProprietaire = null;

    #[ORM\Column(length: 45)]
    private ?string $telProprietaire = null;

    /**
     * @var Collection<int, Bien>
     */
    #[ORM\OneToMany(targetEntity: Bien::class, mappedBy: 'proprietaire')]
    private Collection $biens;

    public function __construct()
    {
        $this->biens = new ArrayCollection();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNomProprietaire(): ?string
    {
        return $this->nomProprietaire;
    }

    public function setNomProprietaire(string $nomProprietaire): static
    {
        $this->nomProprietaire = $nomProprietaire;

        return $this;
    }

    public function getPrenomProprietaire(): ?string
    {
        return $this->prenomProprietaire;
    }

    public function setPrenomProprietaire(string $prenomProprietaire): static
    {
        $this->prenomProprietaire = $prenomProprietaire;

        return $this;
    }

    public function getEmailProprietaire(): ?string
    {
        return $this->emailProprietaire;
    }

    public function setEmailProprietaire(string $emailProprietaire): static
    {
        $this->emailProprietaire = $emailProprietaire;

        return $this;
    }

    public function getTelProprietaire(): ?string
    {
        return $this->telProprietaire;
    }

    public function setTelProprietaire(string $telProprietaire): static
    {
        $this->telProprietaire = $telProprietaire;

        return $this;
    }

    /**
     * @return Collection<int, Bien>
     */
    public function getBiens(): Collection
    {
        return $this->biens;
    }

    public function addBien(Bien $bien): static
    {
        if (!$this->biens->contains($bien)) {
            $this->biens->add($bien);
            $bien->setProprietaire($this);
        }

        return $this;
    }

    public function removeBien(Bien $bien): static
    {
        if ($this->biens->removeElement($bien)) {
            // set the owning side to null (unless already changed)
            if ($bien->getProprietaire() === $this) {
                $bien->setProprietaire(null);
            }
        }

        return $this;
    }

    public function __toString(): string
    {
        return $this->prenomProprietaire . ' ' . $this->nomProprietaire;
    }
}
